testing/test-utils/src/lib/utils/nx-plugin.ts

```diff
@@ -0,0 +1,47 @@
+import { CreateNodes, CreateNodesContext, CreateNodesResult } from '@nx/devkit';
+import { vol } from 'memfs';
+import { MEMFS_VOLUME } from '../constants';
+
+/**
+ * Unit Testing helper for the createNodes function of a Nx plugin.
+ * This function will create files over `memfs` from testCfg.matchingFilesData
+ * and invoke the createNodes function on each of the files provided including potential createNodeOptions.
+ * It will aggregate the results of each invocation and return the CreateNodesResult.
+ *
+ * @example
+ * ```ts
+ * const {projects = {}} = await createFilesAndInvokeCreateNodesOnThem(createNodes, context, undefined, { matchingFilesData});
+ *  // project should have one target created
+ *  const targets = projects[projectRoot]?.targets ?? {};
+ *  expect(Object.keys(targets)).toHaveLength(1);
+ *  // target should be the init target
+ *  expect(targets[`${CP_TARGET_NAME}--init`]).toBeDefined();
+ *  ```
+ *
+ * @param createNodes
+ * @param context
+ * @param createNodeOptions
+ * @param mockData
+ */
+export async function createFilesAndInvokeCreateNodesOnThem<
```

Comment:
This helper is inspired by https://github.com/nrwl/nx/blob/94101643139680f5a9c1072879ca292686db8e3f/packages/js/src/plugins/typescript/plugin.spec.ts#L2178

I did not go with pulling in minimatch as if the implementation changes our tests still pass :D 
WDYT? @Tlacenka

Reply:
I am not an expert on nx plugins so cannot offer advice here.

Reply:
Ok. Thanks anyway for looking.